MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -0,0 +1,100 @@
+
+
+import Foundation
+
+class QuestionFactory: QuestionFactoryProtocol {
+
+    // MARK: - mock data commented for the sprint 7
+    /*
+    private let questions: [QuizQuestion] = [
+        QuizQuestion(image: "The Godfather",
+                     text: "Рейтинг этого фильма больше чем 6?",
+                     correctAnswer: true),
+        QuizQuestion(image: "The Dark Knight",
+                     text: "Рейтинг этого фильма больше чем 6?",
+                     correctAnswer: true),
+        QuizQuestion(image: "Kill Bill",
+                     text: "Рейтинг этого фильма больше чем 6?",
+                     correctAnswer: true),
+        QuizQuestion(image: "The Avengers",
+                     text: "Рейтинг этого фильма больше чем 6?",
+                     correctAnswer: true),
+        QuizQuestion(image: "Deadpool",
+                     text: "Рейтинг этого фильма больше чем 6?",
+                     correctAnswer: true),
+        QuizQuestion(image: "The Green Knight",
+                     text: "Рейтинг этого фильма больше чем 6?",
+                     correctAnswer: true),
+        QuizQuestion(image: "Old",
+                     text: "Рейтинг этого фильма больше чем 6?",
+                     correctAnswer: false),
+        QuizQuestion(image: "The Ice Age Adventures of Buck Wild",
+                     text: "Рейтинг этого фильма больше чем 6?",
+                     correctAnswer: false),
+        QuizQuestion(image: "Tesla",
+                     text: "Рейтинг этого фильма больше чем 6?",
+                     correctAnswer: false),
+        QuizQuestion(image: "Vivarium",
+                     text: "Рейтинг этого фильма больше чем 6?",
+                     correctAnswer: false)
+    ]
+     */
+    private let moviesLoader: MoviesLoading
+    private var movies: [MostPopularMovie] = []
+    weak var delegate: QuestionFactoryDelegate?
```

Comment:
Ошибка при попытке сделать праватным: Property 'delegate' must be declared internal because it matches a requirement in internal protocol 'QuestionFactoryProtocol'. Необходимо убрать переменную из протокола?

Reply:
Понял, да, потому что протокол требует это св-во, поэтому оно не может быть приватным. Оставляем тогда internal)